diff -ur old/contrib/babelfishpg_tsql/antlr/CMakeLists.txt new/contrib/babelfishpg_tsql/antlr/CMakeLists.txt
--- old/contrib/babelfishpg_tsql/antlr/CMakeLists.txt	2023-06-15 11:17:08.000000000 +0000
+++ new/contrib/babelfishpg_tsql/antlr/CMakeLists.txt	2023-07-01 20:10:55.281959729 +0000
@@ -5,14 +5,15 @@
 
 # compiler must be 11 or 14
 set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -O2 -ggdb -w -Wno-deprecated")
-set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -fPIC -O2 -ggdb -w -Wno-deprecated")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -DUSE_UTF8_INSTEAD_OF_CODECVT")
 
 message(STATUS "CMAKE_C_FLAGS=${CMAKE_C_FLAGS}")
+message(STATUS "CMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
 
 # add antrl4cpp artifacts to project environment
 #include_directories($ENV{ANTLR4_RUNTIME_INCLUDE_DIR})
-SET (MYDIR /usr/local/include/antlr4-runtime/)
+SET (MYDIR /usr/include/antlr4-runtime/)
 include_directories(${MYDIR})
 
 # set variable pointing to the antlr tool that supports C++
diff -ur old/contrib/babelfishpg_tsql/Makefile new/contrib/babelfishpg_tsql/Makefile
--- old/contrib/babelfishpg_tsql/Makefile	2023-06-15 11:17:08.000000000 +0000
+++ new/contrib/babelfishpg_tsql/Makefile	2023-07-01 20:10:14.948573518 +0000
@@ -74,10 +74,9 @@
 OBJS += src/linked_servers.o
 OBJS += src/table_variable_mvcc.o
 
-export ANTLR4_JAVA_BIN=java
 export ANTLR4_RUNTIME_LIB=-lantlr4-runtime
-export ANTLR4_RUNTIME_INCLUDE_DIR=/usr/local/include/antlr4-runtime
-export ANTLR4_RUNTIME_LIB_DIR=/usr/local/lib
+export ANTLR4_RUNTIME_INCLUDE_DIR=/usr/include/antlr4-runtime
+export ANTLR4_RUNTIME_LIB_DIR=./
 
 OBJS += src/pltsql_bulkcopy.o
 
@@ -86,9 +85,11 @@
 PG_CXXFLAGS += -Wno-undef -Wall -Wcpp
 PG_CXXFLAGS += -Wno-register # otherwise C++17 gags on PostgreSQL headers
 PG_CXXFLAGS += -I$(ANTLR4_RUNTIME_INCLUDE_DIR)
+PG_CXXFLAGS += -std=c++14 -DUSE_UTF8_INSTEAD_OF_CODECVT
 PG_CFLAGS += -g -Werror
 PG_CFLAGS += -fstack-protector-strong
-PG_CPPFLAGS += -I$(TSQLSRC) -I$(PG_SRC) -DFAULT_INJECTOR
+PG_CFLAGS += -fcommon -Wno-error=format-security -Wno-error=maybe-uninitialized
+PG_CPPFLAGS += -I$(TSQLSRC) -I$(PG_SRC)
 
 SHLIB_LINK += -L$(ANTLR4_RUNTIME_LIB_DIR) $(ANTLR4_RUNTIME_LIB) -lcrypto
 
@@ -167,13 +168,6 @@
 GEN_KEYWORDLIST_DEPS = $(TOOLSDIR)/gen_keywordlist.pl $(TOOLSDIR)/PerfectHash.pm
 TSQLSRC = .
 
-antlr/Makefile: antlr/CMakeLists.txt antlr/TSqlLexer.g4 antlr/TSqlLexer.g4
-	cd antlr && $(cmake) . && cd ..
-
-.PHONY: antlr/libantlr_tsql.a  # to allow CMake's make check the build
-antlr/libantlr_tsql.a: antlr/Makefile
-	$(MAKE) -C $(@D) all
-
 # See notes in src/backend/parser/Makefile about the following two rules
 src/pl_gram.h: src/pl_gram.c
 	touch $@
